src/main/flight/mixer.c

```diff
@@ -613,11 +618,28 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
 
         }
 #endif
+
+        // reduce motorRangeMax when battery is full
+        const float vbatLow = ((float)batteryConfig()->vbatwarningcellvoltage) / 100.0f;
+        const float vbatFull = ((float)batteryConfig()->vbatfullcellvoltage) / 100.0f;
+        const float vBatRange = vbatFull - vbatLow;
+        const float currentCellVoltage = ((float)getBatteryAverageCellVoltage()) / 100.0f; // convert from units of .01V
+        if (motorOutputLimitReduction < 1.0f) {
+            float vBatFactor;
+            // vBatFactor = 0 when voltage is >= vbatFull, and 1 when voltage is <= vbatLow
+            vBatFactor = (vbatFull - MAX(currentCellVoltage, vbatLow)) / vBatRange;
+            if (vBatFactor<0) vBatFactor = 0;
+            motorRangeMaxDecrease = (1.0f - vBatFactor) * motorOutputLimitReduction;
+        }
         currentThrottleInputRange = rcCommandThrottleRange;
         motorRangeMin = motorOutputLow + motorRangeMinIncrease * (motorOutputHigh - motorOutputLow);
-        motorRangeMax = motorOutputHigh;
+        motorRangeMax = motorOutputHigh - motorRangeMaxDecrease * (motorOutputHigh - motorOutputLow);
+        if (debugMode == DEBUG_BATTERY) {
+            debug[2] = lrintf(currentCellVoltage * 100);
```

Comment:
I think debug[1] already contains the currentCellVoltage value.

Reply:
It contains filtered total battery voltage.